rfq/order.go

```diff
@@ -297,6 +297,12 @@ func (c *AssetPurchasePolicy) CheckHtlcCompliance(
 		return fmt.Errorf("error summing asset balance: %w", err)
 	}
 
+	// Due to rounding errors, we may slightly underreport the incoming
+	// value of the asset. So we increase it by exactly one asset unit to
+	// ensure that the fee logic in lnd does not reject the HTLC.
+	roundingCorrection := rfqmath.NewBigIntFromUint64(1)
```

Comment:
Can we always assume the rounding error is less then 1 asset unit?

Reply:
Yes, the imprecision of an integer representation of a division can at most be `0.99999......`, since we're strictly cutting off any fractional part after the division.